[Refactor] Remove storage_format from FE

chaoyli opened the pull request:
The storage_format in TCreateTabletReq is already not-used in BE. So the code paths in FE is also non-sense. I remove the code paths. After the pull request, the `show create table` don't display this property.

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

## Problem Summary(Required) ：
<!-- (Please describe the changes you have made. In which scenarios will this bug be triggered and what measures have you taken to fix the bug?) -->

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

imay commented on be/src/thirdparty/starcache:
why do you delete the starcache, it is needed in the lakehouse scenarios

chaoyli committed:
[Refactor] Remove storage_format from FE
The storage_format in TCreateTabletReq is already not-used in BE.
So the code paths in FE is also non-sense. I remove the code paths.
After the pull request, the `show create table` don't display this property.

chaoyli committed:
[Refactor] Remove storage_format from FE
